glean-core/src/storage/mod.rs

```diff
@@ -31,14 +31,30 @@ impl StorageManager {
         store_name: &str,
         clear_store: bool,
     ) -> Option<JsonValue> {
-        let mut snapshot: HashMap<&str, HashMap<String, JsonValue>> = HashMap::new();
+        let mut snapshot: HashMap<String, HashMap<String, JsonValue>> = HashMap::new();
```

Comment:
To learn a bit more: why is this change needed?

Reply:
Because before we were using a `&'static str`, now we have either a `&'static str` or a `&'a str` (derived from something inside our loop). We need to have a single type (though we might reduce overhead with [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html))